populus/compilation/backends/base.py

```diff
@@ -1,16 +1,39 @@
 import logging
+import os
 
+from eth_utils import (
+    to_tuple,
+)
 from populus.utils.module_loading import (
     get_import_path,
 )
+from populus.utils.filesystem import (
+    recursive_find_files,
+)
 
 
 class BaseCompilerBackend(object):
     compiler_settings = None
+    project_source_extensions = None
+    test_source_extensions = None
 
     def __init__(self, settings):
         self.compiler_settings = settings
         self.logger = logging.getLogger(get_import_path(type(self)))
 
     def get_compiled_contracts(self, source_file_paths, import_remappings):
         raise NotImplementedError("Must be implemented by subclasses")
+
+    @to_tuple
+    def _find_source_files(self, base_dir, extensions):
```

Comment:
I think I would prefer if this was not included on the class and you just directly used `recursive_find_files` in the `get_XXX_paths` methods.